### DISCUSSION TITLE
Attach appsettings.json with Appbuilder

### DISCUSSION THREAD

**Discussion by amuagarwal** [AvaloniaUI/Avalonia#9206] [Q&A] [answered]
Hi 
i am looking for something similar to ConfigureAppConfiguration in .net host builder, please help me before i give up on avalonia

**Comment by timunie** [comment]
Maybe this can help you? https://docs.avaloniaui.net/guides/deep-dives/reactiveui/data-persistence

If that is not what you are looking for please try to add more details about your issue.

**Comment by amuagarwal** [comment]
what i a looking for replacement of

Host.CreateDefaultBuilder()
        .ConfigureAppConfiguration((_, builder) =>
        {
            builder.SetBasePath(Directory.GetCurrentDirectory());
            builder.AddJsonFile("AppSettings.json", optional: false, reloadOnChange: true);
        })
to pass configuration from parent to child objects

**Reply by timunie** [reply]
Normally this should be handled by your ViewModel if I understand that correctly. Are you using the MVVM approach already? Do you need some samples maybe?

**Comment by maxkatz6** [comment, marked as answer]
You don't need Host builder to use appsettings configuration.
https://pradeepl.com/blog/dotnet/configuration-in-a-net-core-console-application/

**Reply by amuagarwal** [reply]
I moved to this approach only with .net core host builder things can be done in one place only and i was looking for same as my application have custom plugin structure where parent is responsible to send configuration base on plugin selection

**Reply by maxkatz6** [reply]
Then you can build your Host with the avalonia app. For example inside of OnFrameworkInitialized
See here user's experience https://github.com/AvaloniaUI/Avalonia/discussions/8588

**Reply by amuagarwal** [reply]
Thanks @maxkatz6 was looking for this only its just a week i start with Avalonia.
